src/hotspot/share/nmt/nmtUsage.cpp

```diff
@@ -52,10 +52,13 @@ void NMTUsage::walk_thread_stacks() {
 }
 
 void NMTUsage::update_malloc_usage() {
-  // Thread critical needed keep values in sync, total area size
+  MallocMemorySnapshot* ms;
+  // Lock needed to keep values in sync, total area size
   // is deducted from mtChunk in the end to give correct values.
-  ThreadCritical tc;
-  const MallocMemorySnapshot* ms = MallocMemorySummary::as_snapshot();
+  {
+    ChunkPoolLocker lock;
+    ms = MallocMemorySummary::as_snapshot();
+  }
```

Comment:
Shortening the scope of the lock seems contrary to the comment explaining its presence.

Reply:
I missed the discussion that suggested this shortening before commenting.  I guess the snapshot
is what provides the atomicity for the remainder, so shortening the scope seems okay after all.